xarray/tests/test_concat.py

```diff
@@ -513,6 +513,14 @@ def test_concat_multiindex(self) -> None:
         assert expected.equals(actual)
         assert isinstance(actual.x.to_index(), pd.MultiIndex)
 
+    def test_concat_along_new_dim_multiindex(self) -> None:
+        x = pd.MultiIndex.from_product([[1, 2, 3], ["a", "b"]])
+        ds = Dataset(coords={"x": x})
+        actual = concat([ds], "new")
+        assert isinstance(actual.x.to_index(), pd.MultiIndex)
+        for k, idx in actual.indexes.items():
+            assert idx is actual.indexes["x"]
```

Comment:
```suggestion
    def test_concat_along_new_dim_multiindex(self) -> None:
        # see https://github.com/pydata/xarray/issues/6881
        x = pd.MultiIndex.from_product([[1, 2, 3], ["a", "b"]])
        ds = Dataset(coords={"x": x})
        concatenated = xr.concat([ds], "new")
        actual = list(concatenated.xindexes.get_all_coords("x"))
        expected = ["x", "a", "b"]
        assert actual == expected
```

nit: the suggestion above is an equivalent test using the new `Indexes` API (`get_all_coords()` checks for index identity). Also, I generally find it helpful to have a link to the issue report for such regression test.